snooty/postprocess.py

```diff
@@ -532,11 +533,34 @@ class TabsSelectorHandler(Handler):
     def __init__(self, context: Context) -> None:
         super().__init__(context)
         self.selectors: Dict[str, List[Dict[str, MutableSequence[n.Text]]]] = {}
+        self.stack: List[str] = []
+        self.scanned_pattern: List[str] = []
+
+    def scan_for_pattern(self, fileid_stack: FileIdStack, node: n.Node) -> None:
+        starting_point = 0
+        target_pattern = ["tabs", "tabs", "procedure"]
+        target_pattern_len = len(target_pattern)
+        if len(self.scanned_pattern) > 0:
+            reverse_list = reversed(self.scanned_pattern)
+            for item in reverse_list:
+                if item == target_pattern[starting_point]:
+                    starting_point += 1
+                if starting_point >= target_pattern_len:
+                    self.context.diagnostics[fileid_stack.current].append(
+                        InvalidNestedTabStructure(
+                            " ".join(reversed(self.scanned_pattern)), node.start[0]
```

Comment:
@rayangler , I can give this a try, but for some reason I was getting an error when trying to reuse the same `reverse_list`. I am still very much a novice to Python. but I can give this a try again see if I am getting any error or if I missed something.

Reply:
I am not 100% sure why I get the different behaviors, but not resuing the `reverse_list` i.e. 

```python
InvalidNestedTabStructure(
                            " ".join(reversed(self.scanned_pattern)), node.start[0]
                        )
```
gives me the desired result

```bash
WARNING(snippets/commands.txt:6ish): Detect tabs that contain tabs that contain procedures: : "tabs tab tabs tab include procedure step"
```

whereas reusing gives i.e. 

```python
InvalidNestedTabStructure(
                            " ".join(reverse_list), node.start[0]
                        )
```

gives me a partial of the list (just the first node

```bash
WARNING(snippets/troubleshooting.txt:6ish): Detect tabs that contain tabs that contain procedures: : "step"
```

Any thoughts?

Reply:
@rayangler I figured it out. Silly me. I guess I needed to turn it back into a list. This approach seems to work 

```python
reverse_list = list(reversed(self.scanned_pattern))
            for item in reverse_list:
                if item == self.target_pattern[starting_point]:
                    starting_point += 1
                if starting_point >= target_pattern_len:
                    self.context.diagnostics[fileid_stack.current].append(
                        InvalidNestedTabStructure(
                            " ".join(reverse_list), node.start[0]
                        )
                    )
                    return
```

Reply:
Admittedly, I'm not a Python expert either. My assumption is that `reverse` returns an iterator, and since it's being used in the `for` loop, it has already gone through certain iterations by the time it's used in the `join`. Please feel free to ignore/keep as-is! I made the assumption that it worked more like JS's `reverse()` function

Reply:
Turning it back into a list makes sense too! Nice